### ISSUE TITLE
Deploy Azure App Service step does not send container download to app

### ISSUE THREAD

**Issue by garrettdass** [OctopusDeploy/Issues#7088]
### Team

- [X] I've assigned a team label to this issue

### Severity

1+ Customer

### Version

Latest

### What happened?

In the `Deploy Azure App Service` step Octopus attempts to download the docker container on Octopus Server instead of initiating the download on the App. 

### Reproduction

1. Create a project with Deploy Azure App Service step:
[Internal Link to Process](https://octopus-operations.octopus.app/app#/Spaces-222/projects/test-az-feeds/deployments/process/steps?actionId=123e12ff-ff70-47fb-a9de-b89cdc88e318)
2. Create an Azure app resource:
[Internal Link to Resource](https://portal.azure.com/#@octopusdeploysandbox.onmicrosoft.com/resource/subscriptions/2db1fb06-5fc0-411b-80e6-0e9bad67d47d/resourcegroups/GD-Repro/providers/Microsoft.Web/sites/GD-Repro/appServices)
3. Attempt to deploy container to webapp and see error:
[Internal Link to Task log](https://octopus-operations.octopus.app/app#/Spaces-222/tasks/ServerTasks-108564/raw)

### Error and Stacktrace

```shell
Task ID:        ServerTasks-108564
Related IDs:    Deployments-4666, Channels-1294, Releases-4780, Projects-1233, Spaces-222, Environments-621
Task status:    Failed
Task queued:    Monday, 27 September 2021 6:48:48 PM +00:00
Task started:   Monday, 27 September 2021 6:48:48 PM +00:00
Task completed: Monday, 27 September 2021 6:48:57 PM +00:00
Task duration:  9 seconds
Server version: 2021.2.7580+Branch.release-2021.2.Sha.db69134412ffbf682ee48e345e2354b7834538ef
Server node:    octopus-i005311-7bb4cbdb4c-mq5wc

                    | == Failed: Deploy Test AZ Feeds release 0.0.3 to Dev ==
18:48:48   Verbose  |   Guided failure is not enabled for this task
18:48:51   Verbose  |   Releasing workers that required packages.
18:48:57   Verbose  |   Successfully called release for dynamic pool lease Leases-37143785 : 21-09-27-1646-7enyb
18:48:57   Fatal    |   The deployment failed because one or more steps failed. Please see the deployment log for details.
                    | 
                    |   == Failed: Acquire packages ==
18:48:48   Info     |     Acquiring packages
18:48:48   Info     |     Making a list of packages to acquire
18:48:48   Info     |     Leasing Windows2019 dynamic worker...
18:48:51   Info     |     Obtained worker lease successfully.
18:48:51   Verbose  |     Leased worker 21-09-27-1646-7enyb from pool Hosted Windows (lease Leases-37143785).
18:48:51   Verbose  |     Package nginx v1.21.3 is required by action 'Deploy an Azure App Service'
18:48:51   Verbose  |     No packages are required on the Octopus Server
18:48:51   Verbose  |     Delta compression is enabled for package transfers from the Octopus Server to deployment targets
18:48:51   Verbose  |     Machine DynamicWorker 21-09-27-1646-7enyb still needs packages nginx v1.21.3 for action ('Deploy an Azure App Service')
18:48:51   Fatal    |     The step failed: Activity failed with error 'The Docker Registry v1 'Docker' at https://index.docker.io does not support downloading packages on the Octopus Server.'.
18:48:51   Verbose  |     Acquire Packages completed
                    |   
                    |     Failed: DynamicWorker 21-09-27-1646-7enyb
18:48:51   Fatal    |       The Docker Registry v1 'Docker' at https://index.docker.io does not support downloading packages on the Octopus Server.
18:48:51   Verbose  |       at Octopus.Core.Packages.DeploymentPackageDownloader.GetPackageFeedWithLocalCacheFor(PackageIdentity identifier, ITaskLog log) in ./source/Octopus.Core/Packages/DeploymentPackageDownloader.cs:line 138
                    |       at Octopus.Core.Packages.DeploymentPackageDownloader.<>c__DisplayClass8_0.<DownloadPackage>b__0() in ./source/Octopus.Core/Packages/DeploymentPackageDownloader.cs:line 41
                    |       at Octopus.Core.Packages.DeploymentPackageDownloader.PackageMemento.GetOrAdd(PackageIdentity identifier, Func`1 factory) in ./source/Octopus.Core/Packages/DeploymentPackageDownloader.cs:line 178
                    |       at Octopus.Server.Orchestration.Targets.Common.PackageStagingMediator.DownloadPackageAndPushToAgent(PackageIdentity packageIdentity, IDeploymentPackageDownloader deploymentPackageDownloader, DeltaCompressionOptions deltaCompressionOptions, PackageCacheOptions packageCacheOptions, PackageAcquisitionExecutionHandlers execs, ITaskLog stepTaskLog, ITaskLog deploymentTargetTaskLog) in ./source/Octopus.Server/Orchestration/Targets/Common/PackageStagingMediator.cs:line 75
                    |       at Octopus.Server.Orchestration.ServerTasks.Deploy.Steps.Acquire.AcquireMachinePackageTask.StagePackage(PackageAcquisitionExecutionHandlers packageAcquisitionHandlers, ITaskLog stepTaskLog, ITaskLog deploymentTargetTaskLog) in ./source/Octopus.Server/Orchestration/ServerTasks/Deploy/Steps/Acquire/AcquireMachinePackageTask.cs:line 148
                    |       at Octopus.Server.Orchestration.ServerTasks.Deploy.Steps.Acquire.AcquireMachinePackageTask.Acquire(AcquiredPackageMap acquiredPackageMap, ITaskContext taskContext, ITaskLog stepTaskLog, ITaskLog deploymentTargetTaskLog) in ./source/Octopus.Server/Orchestration/ServerTasks/Deploy/Steps/Acquire/AcquireMachinePackageTask.cs:line 102
                    |       at Octopus.Server.Infrastructure.EitherAsyncOrSync.Execute(CancellationToken cancellationToken) in ./source/Octopus.Server/Infrastructure/EitherAsyncOrSync.cs:line 38
                    |       at Octopus.Server.Orchestration.ServerTasks.Deploy.Guidance.ExecuteWithoutGuidance(EitherAsyncOrSync callback, String actionName, Boolean actionIsRequiredToRun, ITaskLog taskLog, CancellationToken cancellationToken) in ./source/Octopus.Server/Orchestration/ServerTasks/Deploy/Guidance.cs:line 116
                    |       at Octopus.Server.Orchestration.ServerTasks.Deploy.Guidance.Execute(EitherAsyncOrSync callback, String actionName, Boolean actionIsRequiredToRun, ITaskLog taskLog, Maybe`1 callbackOnExclude, CancellationToken cancellationToken) in ./source/Octopus.Server/Orchestration/ServerTasks/Deploy/Guidance.cs:line 67
                    |       at Octopus.Server.Orchestration.ServerTasks.Deploy.ExecutionTaskController`1.<>c__DisplayClass29_3.<<AcquirePackages>b__4>d.MoveNext() in ./source/Octopus.Server/Orchestration/ServerTasks/Deploy/ExecutionTaskController.cs:line 360
                    |       --- End of stack trace from previous location ---
                    |       at Octopus.Server.Orchestration.ServerTasks.Deploy.ExecutionTaskController`1.ExecuteWithTransientErrorDetection(Func`2 action, ITaskLog taskLog, CancellationToken cancellationToken, DeploymentTarget deploymentTarget) in ./source/Octopus.Server/Orchestration/ServerTasks/Deploy/ExecutionTaskController.cs:line 753
                    |       at Octopus.Server.Orchestration.ServerTasks.Deploy.ExecutionTaskController`1.<>c__DisplayClass29_2.<<AcquirePackages>b__3>d.MoveNext() in ./source/Octopus.Server/Orchestration/ServerTasks/Deploy/ExecutionTaskController.cs:line 363
                    |       --- End of stack trace from previous location ---
                    |       at Octopus.Server.Infrastructure.Orchestration.UnitsOfWork.UnitOfWorkExecutor.<>c__DisplayClass4_0`2.<<Execute>b__0>d.MoveNext() in ./source/Octopus.Server/Infrastructure/Orchestration/UnitsOfWork/UnitOfWorkExecutor.cs:line 74
                    |       --- End of stack trace from previous location ---
                    |       at Octopus.Core.Infrastructure.UnitsOfWork.UnitOfWorkExtensionMethods.DoAsync(IUnitOfWork unitOfWork, Func`1 action, CancellationToken cancellationToken, String name) in ./source/Octopus.Core/Infrastructure/UnitsOfWork/UnitOfWorkExtensionMethods.cs:line 73
                    |       at Octopus.Core.Infrastructure.UnitsOfWork.UnitOfWorkExtensionMethods.DoAsync(IUnitOfWork unitOfWork, Func`1 action, CancellationToken cancellationToken, String name) in ./source/Octopus.Core/Infrastructure/UnitsOfWork/UnitOfWorkExtensionMethods.cs:line 73
                    |       at Octopus.Server.Infrastructure.Orchestration.UnitsOfWork.UnitOfWorkExecutor.Execute[T1,T2](Func`4 action, CancellationToken cancellationToken, String name) in ./source/Octopus.Server/Infrastructure/Orchestration/UnitsOfWork/UnitOfWorkExecutor.cs:line 75
                    |       at Octopus.Server.Orchestration.ServerTasks.Deploy.ExecutionTaskController`1.<>c__DisplayClass29_1.<<AcquirePackages>b__2>d.MoveNext() in ./source/Octopus.Server/Orchestration/ServerTasks/Deploy/ExecutionTaskController.cs:line 364
                    |       --- End of stack trace from previous location ---
                    |       at Octopus.Server.Orchestration.ServerTasks.ParallelWorkOrderAsync`1.<>c__DisplayClass7_1.<<Execute>b__4>d.MoveNext() in ./source/Octopus.Server/Orchestration/ServerTasks/ParallelWorkOrderAsync.cs:line 57
                    |       --- End of stack trace from previous location ---
                    |       at Octopus.Server.Infrastructure.EitherAsyncOrSync.Execute(CancellationToken cancellationToken) in ./source/Octopus.Server/Infrastructure/EitherAsyncOrSync.cs:line 38
                    |       at Octopus.Server.Orchestration.ServerTasks.OctoThreadClosureAsync`1.Execute(CancellationToken cancellationToken) in ./source/Octopus.Server/Orchestration/ServerTasks/OctoThreadClosureAsync.cs:line 60
                    |     
                    |       Failed: Upload package nginx v1.21.3
                    |       
                    |   == Skipped: Step 1: Deploy an Azure App Service ==
18:48:51   Info     |     Step "Deploy an Azure App Service" runs only when all previous steps succeeded; skipping
```

### More Information

[Internal Link](https://help.octopus.com/t/question-deploy-azure-app-service-step/27195/3)

### Workaround

You can correct this behaviour by going into the offending Azure App Service step, and flipping the radio button selection under the "Packages" configuration from “Deploy from a container image” to “Deploy from a zip…” and then back to “Deploy from a container image”. This will ensure the correct configuration is persisted, and the step should function as expected from there.

**Comment by andrewabest** [comment]
Release note: Fixes an issue with the Azure App Service step where package aquisition would fail when configured with a container

**Comment by Octobob** [comment]
:tada: The fix for this issue has been released in:

| Release stream | Release     |
| -------------- | ----------- |
| 2021.3 | 2021.3.6093 |

**Comment by Octobob** [comment]
:tada: The fix for this issue has been released in:

| Release stream | Release     |
| -------------- | ----------- |
| 2021.2 | 2021.2.7693 |
| 2021.3 | 2021.3.6093 |
| 2022.1+ | _all releases_ |

**Comment by Octobob** [comment]
:tada: The fix for this issue has been released in:

| Release stream | Release     |
| -------------- | ----------- |
| 2022.2 | 2022.2.0 |

**Comment by Octobob** [comment]
:tada: The fix for this issue has been released in:

| Release stream | Release     |
| -------------- | ----------- |
| 2021.2 | 2021.2.7693 |
| 2021.3 | 2021.3.6093 |
| 2022.1+ | _all releases_ |

**Comment by NickSchSp** [comment]
I just encountered this issue on version 2022.3.4954
The way to reproduce is different.

1. Create a step for Deploy To Azure App Service and configure container as package.
2. This will run fine.
3. Clone the step create in 1 and change the package.
4. When running the issue occurs where Octopus tries to download the image on the server instead of just setting the metadata on the App Service.

As a workaround I created a new step and copied all values manually. Not a big deal but a small inconvenience.
